app/scripts/components/exploration/components/datasets/colormap-options.tsx

```diff
@@ -0,0 +1,227 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import * as d3 from 'd3-scale-chromatic';
+import * as d3Lib from 'd3';
+import { Icon } from "@trussworks/react-uswds";
+
+import { themeVal } from '@devseed-ui/theme-provider';
+import { variableGlsp } from '$styles/variable-utils';
+
+import './colormap-options.scss';
+
+const Container = styled.div`
+  box-shadow: ${themeVal('boxShadow.elevationB')};
+  width: 383px;
+  overflow-y: auto;
+  max-height: 500px;
+`;
+
+const Header = styled.div`
+`;
+
+const ColormapWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const ColormapItem = styled.div<{ selected: boolean }>`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  cursor: pointer;
+  padding: ${variableGlsp(0.5)};
+  border-bottom: 1px solid ${themeVal('color.base-200')};
+  border-radius: ${themeVal('shape.rounded')};
+  gap: 12px;
+
+  &:hover {
+    background-color: #f4f4f4;
+  }
+
+  ${({ selected }) =>
+    selected &&
+    `
+    border-radius: 0;
+    outline: 2px solid #007BFF;
+    outline-offset: -2px;
+    background-color: #E8F0FD;
+    font-weight: 600;
+  `}
+`;
+
+const ColormapLabel = styled.div`
+  text-align: left;
+  flex: 1;
+`;
+
+interface ColormapPreviewProps {
+  colormap: string[];
+}
+
+export const ColormapPreview = styled.span<ColormapPreviewProps & { width?: string }>`
+  width: ${({ width }) => width ?? '260px'};
+  height: 12px;
+  background: ${({ colormap }) =>
+    `linear-gradient(to right, ${colormap.join(', ')})`};
+  border-radius: ${themeVal('shape.rounded')};
+  border: 1px solid ${themeVal('color.base-200')};
+  display: flex;
+  cursor: default;
+`;
+
+const ToggleContainer = styled.div`
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+  border-top: 1px solid ${themeVal('color.base-200')};
+  border-bottom: 1px solid ${themeVal('color.base-200')};
+`;
+
+const ToggleLabel = styled.label``;
+
+const ToggleInput = styled.input.attrs({ type: 'checkbox' })`
+  display: none;
+`;
+
+export const SEQUENTIAL_COLORMAPS = [
```

Comment:
🤔 Should we have a step  to validate these colormaps are supported color maps?( from https://openveda.cloud/api/raster/colorMaps)? In reality, the supported colormap will change so rarely that I am not sure if we need to add this one more step, but then we will be safe from whichever update from the backend.

Reply:
Ah, I was hoping that we were doing that - fetching the supported colormaps from TiTiler instead of hard-coding in the client?

- https://github.com/NASA-IMPACT/veda-ui/issues/885

Reply:
Ah, I see; I missed that there is even an endpoint for getting the colors of each colormap using each colormap ID - ex. https://openveda.cloud/api/raster/colorMaps/magma. Then it might make sense to have only IDs for colormaps on the client side and get the values from the endpoint.

Reply:
I wasn't aware of the endpoint as well, thanks @j08lue!

@hanbyul-here Since we use d3-color-schematic already which supports the colorMaps by ID and to avoid sending requests on each selection of a colorMap to fetch the values, I don't think we need to hardcode them at all in the UI. In general, we could fetch them from titiler, list them in the UI and use d3-color-schematic to interpolate the values. What do you think about this?

Reply:
Oh, don't we want to curate the colormaps (not all the colormap options from the colormap endpoint)? I meant the IDs for these curated colormaps can be on the front end. (Sorry for not being clear.) 

I think it is always best to have one source of truth, so we don't have to think about how the colors from Titiler don't seem to match 100% with what the legend shows. (If that ever happens -  So we probably should at least check where D3 colormap values are from and how the values match up with the titiler values.) I also understand your concerns about not making too many requests. Since this is the type of data that doesn't change very often, I wonder if we can make a request while building and save it as a local file? 🤔 (This sounds like a way to make this ticket go out of the scope 😅 )

Reply:
So titiler uses colorMaps from rio-tiler. And rio-tiler uses colorMaps (at least the ones that are currently on the client-side) derived from Matplotlib: https://cogeotiff.github.io/rio-tiler/colormap/#colormaps

d3-scale-chromatic also [derives the ones we use from Matplotlib](https://www.npmjs.com/package/d3-scale-chromatic/v/1.5.0#interpolateMagma), which is why I think we will add unnecessary overhead if we send requests to fetch hexes per colorMap ID or save them in local files.

> Oh, don't we want to curate the colormaps (not all the colormap options from the colormap endpoint

Yes I think this was the idea, to curate a selection of most widely used colors instead of showing all the supported ones. Users could still pass in a valid colorMap name which is not in the list at the moment (e.g. PuRd) and it should show up fine as a default value.

Reply:
I see the pros and cons of all options. I would really like a solution that makes maximum use of the nice TiTiler built-ins. But the decision is up to you! 

I lined up the options here, for overview (using an [ADR template](https://github.com/adr/madr/blob/develop/template/adr-template.md?plain=1)):

- https://github.com/NASA-IMPACT/veda-ui/issues/885#issuecomment-2296121564

Please feel free to review and let me know if I should open a proper ADR PR.